ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -215,6 +216,12 @@ void start() throws InterruptedException {
     } catch (StoreException e) {
       logger.error("Error while starting the DiskManager for {} ; no stores will be accessible on this disk.",
           disk.getMountPath(), e);
+      // Set the state of the disk to UNAVAILABLE. This will prevent new replicas to be bootstrapped on this disk.
+      // TODO: We might need to use DiskFailureHandler to immediately reset existing replicas on this disk and reduce
+      //  instance capacity so that existing replicas are reassigned to new hosts and Helix doesn't assign more replicas
+      //  than the host can handle.
+      logger.info("Setting disk {} as UNAVAILABLE locally", disk.getMountPath());
+      disk.setState(HardwareState.UNAVAILABLE);
```

Comment:
this is probably not going to work since it will be overridden by the HelixClusterManager once there is a DataNodeConfig update.